File: app/src/main/java/net/squanchy/favorites/view/FavoritesListView.java
package net.squanchy.favorites.view;

import android.content.Context;
import android.support.annotation.Nullable;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.AttributeSet;

import net.squanchy.R;
import net.squanchy.schedule.domain.view.Schedule;
import net.squanchy.schedule.view.ScheduleViewPagerAdapter;
import net.squanchy.support.view.CardSpacingItemDecorator;

public class FavoritesListView extends RecyclerView {

    private FavoritesAdapter adapter;

    public FavoritesListView(Context context) {
        super(context);
    }

    public FavoritesListView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public FavoritesListView(Context context, @Nullable AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
    }

    @Override
    protected void onFinishInflate() {
        super.onFinishInflate();

        LinearLayoutManager layoutManager = new LinearLayoutManager(getContext());
        setLayoutManager(layoutManager);
        adapter = new FavoritesAdapter(getContext());
        setAdapter(adapter);

        int horizontalSpacing = getResources().getDimensionPixelSize(R.dimen.card_horizontal_margin);
        int verticalSpacing = getResources().getDimensionPixelSize(R.dimen.card_vertical_margin);
        addItemDecoration(new CardSpacingItemDecorator(horizontalSpacing, verticalSpacing));
    }

    public void updateWith(Schedule newData, ScheduleViewPagerAdapter.OnEventClickedListener listener) {
        adapter.updateWith(newData, listener);
    }
}

File: app/src/main/java/net/squanchy/favorites/view/FavoritesListView.kt
package net.squanchy.favorites.view

import android.content.Context
import android.util.AttributeSet
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import net.squanchy.R
import net.squanchy.support.view.CardSpacingItemDecorator
import net.squanchy.support.view.setAdapterIfNone

internal class FavoritesListView @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyle: Int = 0
) : RecyclerView(context, attrs, defStyle) {

    private val adapter = FavoritesAdapter(context)

    override fun onFinishInflate() {
        super.onFinishInflate()

        val layoutManager = LinearLayoutManager(context)
        setLayoutManager(layoutManager)

        val horizontalSpacing = getResources().getDimensionPixelSize(R.dimen.card_horizontal_margin)
        val verticalSpacing = getResources().getDimensionPixelSize(R.dimen.card_vertical_margin)
        addItemDecoration(CardSpacingItemDecorator(horizontalSpacing, verticalSpacing))
    }

    fun updateWith(newData: List<FavoritesItem>, showRoom: Boolean, listener: OnFavoriteClickListener) {
        setAdapterIfNone(adapter)
        adapter.updateWith(newData, showRoom, listener)
    }
}
